nebari-dev/nebari
# Use tofu binary instead of terraform one

## Reference Issues or PRs
Closes #2762



## What does this implement/fix?

_Put a `x` in the boxes that apply_


## Testing


## How to test this PR?

I think there are two important things to test with this PR: (1) deploy from scratch using the OpenTofu binary, and (2) upgrade an existing cluster using the OpenTofu binary. To test:

1. Deploy Nebari off of [this branch](https://github.com/nebari-dev/nebari/tree/terraform-to-opentofu), either locally or to any of the supported cloud providers.
2. Upgrade an existing Nebari deployment using [this branch](https://github.com/nebari-dev/nebari/tree/terraform-to-opentofu) and redeploy

All resources should be correctly deployed and Nebari should be running as usual after.



## Any other comments?

This has been successfully tested in the following scenarios:

* [Local deployment](https://github.com/nebari-dev/nebari/actions/runs/11379267190/job/31659642601?pr=2773)
* [AWS deployment](https://github.com/nebari-dev/nebari/actions/runs/11379153520)
* [GCP deployment](https://github.com/nebari-dev/nebari/actions/runs/11380055783)
* Upgrade of existing `2024.9.1` GCP deployment to https://github.com/nebari-dev/nebari/pull/2773/commits/8f709dacce5d22774b6895a8207e9d81b8251102
